pubsub/gcp/pubsub/pubsub.go

```diff
@@ -221,19 +221,19 @@ func (g *GCPPubSub) Subscribe(req pubsub.SubscribeRequest, daprHandler func(msg
 	topic := g.getTopic(req.Topic)
 	sub := g.getSubscription(g.metadata.consumerID + "-" + req.Topic)
 
-	go g.handleSubscriptionMessages(topic, sub, daprHandler)
+	go g.handleSubscriptionMessages(topic, sub, handler)
 
 	return nil
 }
 
-func (g *GCPPubSub) handleSubscriptionMessages(topic *gcppubsub.Topic, sub *gcppubsub.Subscription, daprHandler func(msg *pubsub.NewMessage) error) error {
+func (g *GCPPubSub) handleSubscriptionMessages(topic *gcppubsub.Topic, sub *gcppubsub.Subscription, handler pubsub.Handler) error {
 	err := sub.Receive(context.Background(), func(ctx context.Context, m *gcppubsub.Message) {
 		msg := &pubsub.NewMessage{
 			Data:  m.Data,
 			Topic: topic.ID(),
 		}
 
-		err := daprHandler(msg)
+		err := handler(ctx, msg)
```

Comment:
I haven't looked too much into the gcp pubsub sdk so assuming it's safe to use the provided context here for the handler.